src/plugins/console/public/application/containers/editor/monaco/use_resize_checker_utils.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useRef } from 'react';
+import { ResizeChecker } from '@kbn/kibana-utils-plugin/public';
+import { monaco } from '@kbn/monaco';
+
+/**
+ * Hook that returns functions for setting up and destroying a {@link ResizeChecker}
+ * for a Monaco editor.
+ */
+export const useResizeCheckerUtils = () => {
+  const resizeChecker = useRef<ResizeChecker | null>(null);
+
+  const setupResizeChecker = (
+    divElement: HTMLDivElement,
+    editor: monaco.editor.IStandaloneCodeEditor
+  ) => {
+    resizeChecker.current = new ResizeChecker(divElement);
```

Comment:
suggest to check if there is already a resizeChecker.current and destroy it if there is one

Reply:
Thanks for the review @Dosant! Added this check with https://github.com/elastic/kibana/pull/180494/commits/f02739bb9bf11e2023296e62d7361b91db4d887d.